assets/src/components/site-scan-context-provider/get-slugs-from-validation-results.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * Internal dependencies
+ */
+import { getPluginSlugFromFile } from '../../common/helpers/get-plugin-slug-from-file';
+
+/**
+ * Retrieve slugs of plugins and themes from a list of validation results.
+ *
+ * See the corresponding PHP logic in `\AMP_Validated_URL_Post_Type::render_sources_column()`.
+ *
+ * @param {Object[]} validationResults
+ * @return {Object} An object consisting of `pluginSlugs` and `themeSlugs` arrays.
+ */
+export function getSlugsFromValidationResults( validationResults = [] ) {
+	const plugins = new Set();
+	const themes = new Set();
+
+	for ( const result of validationResults ) {
+		if ( ! result.sources ) {
+			continue;
+		}
+
+		for ( const source of result.sources ) {
+			if ( source.type === 'plugin' ) {
+				plugins.add( getPluginSlugFromFile( source.name ) );
```

Comment:
```suggestion
				const pluginSlug = getPluginSlugFromFile( source.name );
				if ( 'gutenberg' !== pluginSlug || result.sources.length === 1 ) {
					plugins.add( getPluginSlugFromFile( source.name ) );
				}
```

Reply:
That's right. We can also reuse the already retrieved `pluginSlug` when adding a slug to the set:
```suggestion
				const pluginSlug = getPluginSlugFromFile( source.name );
				if ( 'gutenberg' !== pluginSlug || result.sources.length === 1 ) {
					plugins.add( pluginSlug );
				}
```